### PR TITLE
CreateCurriculum class skeletons

### LINKED ISSUE #73: [Subissue 10.1] Use Case skeleton classes

### PR DESCRIPTION
**Pull request by MMachadoUofT**
This is not yet a functional use case: this was simply to create the class structure with method signatures so everyone who needed access to this code was aware of how it would work.

### CODE REVIEW

**Review by bmguiler — APPROVED** [review]
Follows Clean Architecture perfectly and ready to be implemented when you get the chance

**Review by Bhan-Aayush — APPROVED** [review]
It follows clean architecture perfectly. It will help me with my use cases.

File: src/main/java/use_cases/create_curriculum/CreateCurriculumController.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    /**
+     * The controller method for creating a new curriculum belonging to the active user
+     *
+     * @param curriculumName the new Curriculum's name
+     */
+    public void createCurriculum(String curriculumName) {
+
+    }
+
```
Comment (Amir1441) [inline src/main/java/use_cases/create_curriculum/CreateCurriculumController.java:20]: It seems like you correctly followed the rules and regulations of CA here

File: src/main/java/use_cases/create_curriculum/CreateCurriculumInteractor.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    /**
+     * Executes the use case for creating a new Curriculum
+     *
+     * @param name the new Curriculum's name
+     */
+    @Override
+    public void createCurriculum(String name) {
+
+    }
+
```
Comment (Amir1441) [inline src/main/java/use_cases/create_curriculum/CreateCurriculumInteractor.java:21]: Good thing going with your descriptions. It helps anyone to understand the broader view of everything.

**Review by Amir1441 — APPROVED** [review]
Overall the structure does seem solid. A functional usecase is probably the next step for you. This will in turn allow others in the group to access the code and make use of it accordingly. The only thing that's left is actually implementing the skeletons but overall, you are on the right path.

File: src/main/java/use_cases/create_curriculum/CreateCurriculumInputBoundary.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * @author MMachadoUofT
+ */
+public interface CreateCurriculumInputBoundary {
+
+    /**
+     * Executes the use case for creating a new Curriculum
+     *
+     * @param name the new Curriculum's name
+     */
+    void createCurriculum(String name);
```
Comment (Bhan-Aayush) [inline src/main/java/use_cases/create_curriculum/CreateCurriculumInputBoundary.java:17]: Correct variable type and appropriate name for the curriculum's name.

**Pull request merged**